Eliminate sex ratio application, add vignette, and update the process for writeComps

chantelwetzel-noaa opened the pull request:
This pull request includes the following changes:
- Removes the sex ratio
- Sets data that cannot be expanded due to missing sample weights to a weights of 1.  This sets the expansions for those records to 1 similar to records with NAs (missing trip weights). This primarily impacts WA records which often do not have EXP_WT or RWT_LBS.  Since this has a large impact on the expanded composition data, this change should be reviewed carefully.
- Implements a vignette to show how sablefish data are worked up.
- Writes the composition data in a form that is more easily input into SS3 models and includes a bug fix allowing compositions to be created across a variety of data binning.
- Simplify function arguments:
  - writeComps now uses `comp_bins` for both creating length and age composition data.
  - deprecate `plot` input across functions and replace with `savedir` which allows greater control of creating and saving figures.

iantaylor-NOAA commented on R/get_pacfin_expansions.R:
```diff
@@ -0,0 +1,211 @@
+      )
+    }
+  }
+  fa = weight_length_estimates |>
```
I see a few places where this function doesn't match the standard style (e.g. `=` vs `<-`), but the github action will take care of that.

chantelwetzel-noaa replied:
Good catch.  I will fix this.

iantaylor-NOAA commented on R/get_pacfin_expansions.R:
```diff
@@ -0,0 +1,211 @@
+#' Expand bds data up to the tow and catch level
```
I would capitalize BDS here.

iantaylor-NOAA commented on vignettes/sablefish_example.Rmd:
```diff
@@ -0,0 +1,175 @@
+species_code <- "SABL"
+
+# Files
+file_bds <- fs::path("..", "PacFIN.SABL.bds.11.Nov.2024.RData")
```
I worked through an earlier version of this vignette using the file \\\\nwcfile\FRAM\Assessments\CurrentAssessments\sablefish_2025\data-raw\PacFIN.SABL.bds.17.Jul.2023.RData and the corresponding catch file, but I don't see anything like data_commercial_catch_expansions.csv on the network to work through the file.

However, the changes since a few weeks ago aren't that large and all look good so I don't need to run the code again.

iantaylor-NOAA commented on vignettes/sablefish_example.Rmd:
````diff
@@ -0,0 +1,175 @@
+  )
+```
+
+`cleanPacFIN()` filters out data and adds additional columns that are used in the data processing.  One of the columns added, `fleet`, is used to separate the data by model fleet. The default values in `fleet` are set to the gear names defined by `keep_gears` in `cleanPacFIN()`.  The values in this column should be modified the user, if necessary.
````
Add "by" to "modified the user". Alternatively, the sentence could be revised to note that the user could either modify the fleet values coming out of `cleanPacFIN()` or modify the fleet values in the data frame created by `getComps()` (or maybe `writeComps()`).

iantaylor-NOAA reviewed (approved):
@chantelwetzel-noaa, all these changes look great. The addition of the `get_pacfin_expansions()` and the vignette are big improvements to the usability of the package.

@kellijohnson-NOAA thank you for all your work on this PR and the successful handoff to @chantelwetzel-noaa.

Most of my comments are really trivial, especially the stuff related to dashes and hyphens. I think that `--` should be used only in roxygen comments because they get rendered, but not in comments or other occurrences in the code but I'm not sure it's worth the effort to clean that up.

Finally, I pushed an additional commit to this package adding authorship to a bunch of functions where it was clearly incomplete or missing. I'm sure I missed a few places, but I think it's a bit more accurate now.

chantelwetzel-noaa commented on R/EF1_Numerator.R:
```diff
@@ -28,26 +28,40 @@
 
   if (verbose) {
-    cat("\nSampled pounds per trip:\n\n")
+    cli::cli_inform("\nSampled pounds per trip:\n\n")
```
Good catch. I will fix this.

chantelwetzel-noaa commented on R/formatCatch.R:
```diff
@@ -38,6 +38,8 @@
 #'
 formatCatch <- function(catch,
                         strat,
+                        # TODO: yearname and valuename should be year_name and
```
This is an older note and not something I considered.  I think standardizing how arguments are specified would be an improvement to the package but not something I want to do right now just to be safe.

chantelwetzel-noaa commented on R/getExpansion_1.R:
```diff
@@ -1,96 +1,133 @@
-#' @seealso Called within this function are
-#' \code{\link{EF1_Numerator}} \code{\link{EF1_Denominator}}.
-#' Called after this function should be \code{\link{getExpansion_2}}.
+#' @seealso
```
I didn't add this here because I don't consider it necessary upstream but rather this new function as a wrapper.  I don't know what the right approach is for this type of function.

chantelwetzel-noaa commented on R/get_pacfin_expansions.R:
```diff
@@ -0,0 +1,211 @@
+          data_exp1[, "stratification"] <- data_exp1[, stratification.cols]
+        } else {
+          separate <- unique(gsub(
+            "^[a-zA-Z]+(\\s*[[:punct:]]\\s*)[a-zA-Z]+$",
```
Haha!  You think I have any idea what this is doing?!  This is pulled directly from `getExpansion_2` since it was needed to pass to that function in this wrapper approach.  I agree that long-term we should dig into these types of expressions to get better at them and add notes on how they work, or replace them.

iantaylor-NOAA replied:
I added some explanatory comments about what I think is going on in ff4a295

chantelwetzel-noaa commented:
@iantaylor-NOAA I think I have addressed all of your comments.  Additionally, I have added the minor changes in the other pull request into this branch.  Please look over where this pull request stands and let me know if additional revisions are needed or if it can be merged.
